### ISSUE TITLE
"value assigned to variable XXX is unused" with loop

### ISSUE THREAD

**Issue by daurnimator** [mpeterv/luacheck#18]
Incorrect warning with following code?
```lua
local function addone(n)
    return n+1
end
local a = 0
while true do
    a = addone(a)
end
```

**Comment by mpeterv** [comment]
Which version of luacheck are you using? This was indeed a bug versions 0.7.2-0.7.3 but it should be fixed in luacheck 0.8.0, at least I can't reproduce it.

**Comment by daurnimator** [comment]
ah ha, I am using 0.7.3.
